NebulaWorld/SimulatedWorld.cs

```diff
@@ -227,10 +227,17 @@ public static void OnGameLoadCompleted()
             // Assign our own color
             UpdatePlayerColor(LocalPlayer.PlayerId, LocalPlayer.Data.Color);
 
-            // TODO: Investigate where are the weird position coming from ?
-            // GameMain.mainPlayer.transform.position = data.Position.ToUnity();
-            // GameMain.mainPlayer.transform.eulerAngles = data.Rotation.ToUnity();
-
+            // Change player location from spawn to the last known
+            VectorLF3 UPosition = new VectorLF3(LocalPlayer.Data.UPosition.x, LocalPlayer.Data.UPosition.y, LocalPlayer.Data.UPosition.z);
+            if (UPosition != VectorLF3.zero)
+            {
+                GameMain.data.ArrivePlanet(GameMain.data.galaxy.PlanetById(LocalPlayer.Data.LocalPlanetId));
+                GameMain.mainPlayer.planetId = LocalPlayer.Data.LocalPlanetId;
+                GameMain.mainPlayer.position = LocalPlayer.Data.LocalPlanetPosition.ToUnity(); ;
+                GameMain.mainPlayer.uPosition = UPosition;
+                GameMain.mainPlayer.uRotation = Quaternion.Euler(LocalPlayer.Data.Rotation.ToUnity());
+            }
```

Comment:
This seems wrong, you are always calling `ArrivePlanet()` even if the player is in space, shouldn't you guard it with a check to the `LocalPlanetId`?
Also isn't the `UPosition` always greater then 0,0,0 since it is the player Position in the Universe.

Reply:
Now I am thinking about it and should move all of this to the handshake instead of having it in `OnGameLoadCompleted()`. The issue is, that client is joining the server with ` DSPGame.StartGameSkipPrologue(gameDesc);` after handshake is received. Which automatically spawns the player at the landing site by default. So the client will request the factory of the first planet and after everything is loaded it will move him to a different planet and request again the next factory. 
So I will move this code to the handshake so the proper factory is requested at first.
 
To your points:

- You are right about the `ArrivePlanet()`, I left it there is since there is already a check inside for this. But I will add the extra check so it will look better.
- In this case, `UPosition` is the latest position of the player and it is received from the server. So it will be always 0,0,0 when player connects for the first time. Client automatically spawns player at the spawn position based on the `GameDesc` data, so there is no need to change player position in that case.